openlibrary/macros/AvailabilityButton.html

```diff
@@ -0,0 +1,79 @@
+$def with(page, availability)
+
+$ waiting_loan = ctx.user and ctx.user.get_waiting_loan_for(page)
+$ user_loan = None
+$ my_turn_to_borrow = waiting_loan and waiting_loan['status'] == 'available' and waiting_loan['position'] == 1
+$ wlsize = availability.get('num_waitlist', 0)
+$ current_and_available_loans = []
+
+$if page.get('ocaid'):
+  $ current_and_available_loans = page.get_current_and_available_loans()
+  $ user_loan = None
+  $if ctx.user:
+    $for current_loan in current_and_available_loans[0]:
+      $if current_loan['user'] == ctx.user.key:
+        $ user_loan = current_loan
+        $ break
+
+$ current_loans = current_and_available_loans[0] if current_and_available_loans else []
+
+$if ctx.user:
+    $for current_loan in current_loans:
+        $if current_loan['user'] == ctxuser.key:
+            $ user_loan = current_loan
+            $ break
+
+$if availability['status'] in ['open', 'borrow_available', 'borrow_unavailable']:
+  $ read_link = '/borrow/ia/%s' % availability['identifier']
+
+  $if user_loan:
+    <a href="$read_link" title="Borrow"
+       data-userid="$(user_loan['userid'])" id="read_ebook"
+       class="borrow-btn borrow-link">Read eBook</a>
+
+    $ return_url = page.url.rsplit('/', 1)[0] + '/do_return/borrow'
```

Comment:
QUESTION: How fragile is this text parsing, i.e. when using different kinds of base URLS, e.g. with subdomains, localhost, with/without protocol, etc.?